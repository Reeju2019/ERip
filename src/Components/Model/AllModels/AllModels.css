.allmodels {
  background-color: #f5f6fa;
}

.modelCard {
  background-color: white;
  width: 11vmax;
  padding: 1.5vmax;
  cursor: pointer;
}

.modelImg {
  width: 7vmax;
  height: 7vmax;
  display: flex;
  margin: auto;
}

.modelName {
  font-size: 1.2vmax;
  font-weight: bold;
  display: flex;
  margin-top: 10px;
  justify-content: center;
}
