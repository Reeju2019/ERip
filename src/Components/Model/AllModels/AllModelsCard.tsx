import * as React from 'react'
import { useNavigate } from 'react-router-dom'

interface IAllModelsCardProps {
  models: {
    id: string
    modelName: string
    image: string
  }
}

const AllModelsCard: React.FunctionComponent<IAllModelsCardProps> = ({ models }) => {
  const navigate = useNavigate()
  const modelClick = () => {
    localStorage.setItem('modelId', JSON.stringify(models.id))
    const device: string = localStorage.getItem('deviceId') as string
    const brand: string = localStorage.getItem('brandId') as string
    const deviceId = JSON.parse(device)
    const brandId = JSON.parse(brand)
    navigate(`/${deviceId}/${brandId}/${models.id}`)
  }
  return (
    <>
      <div onClick={modelClick} className='modelCard'>
        <div>
          <img src={models.image} className='modelImg' alt={models.modelName} />
        </div>
      </div>
      <p className='modelName'>{models.modelName}</p>
    </>
  )
}

export default AllModelsCard
