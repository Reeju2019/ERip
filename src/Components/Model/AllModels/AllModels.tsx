import React from 'react'
import { Container } from 'react-bootstrap'
import AllModelsCard from './AllModelsCard'
import './AllModels.css'
import ModelData from '../../../Data/Product.mock.json'
import React, { useEffect, useState } from 'react'
import { useParams } from 'react-router-dom'

interface Service {
  serviceImage: string
  service_name: string
  price: number
  off: number
  repairTime: string
  warranty: string
  repairType: string
}

interface Model {
  id: string
  model_name: string
  image: string
  service: Service[]
}

const AllModels: React.FunctionComponent = () => {
  const { deviceId, brandId } = useParams()

  const [modelData, setModelData] = useState<Model[]>([])
  useEffect(() => {
    localStorage.setItem('brandId', JSON.stringify(brandId))
    ModelData.device.map((deviceItem) => {
      if (deviceItem.device_name === deviceId) {
        deviceItem.brand.map((brandItem) => {
          if (brandItem.brand_name === brandId) {
            setModelData(brandItem.model)
          }
        })
      }
    })
  }, [])

  return (
    <>
      <section className='allmodels'>
        <Container className='d-flex flex-wrap justify-content-center'>
          {modelData.map((item, index) => {
            return (
              <div key={index} className='m-3'>
                <AllModelsCard models={item} />
              </div>
            )
          })}
        </Container>
      </section>
    </>
  )
}

export default AllModels
