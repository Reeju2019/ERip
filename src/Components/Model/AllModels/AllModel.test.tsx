import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import AllModels from './AllModels'

test('all model', () => {
    render(<AllModels />)
})
describe('all model rendering specification', () => {
    it('SnapShot testing', () => {
        const component = renderer.create(<AllModels />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('render without crashing all model', () => {
        const div = document.createElement('div')
        render(<AllModels />)
        ReactDOM.unmountComponentAtNode(div)
    })
})
