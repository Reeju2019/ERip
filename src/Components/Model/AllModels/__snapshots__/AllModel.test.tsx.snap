// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all model rendering specification SnapShot testing 1`] = `
<section
  className="allmodels"
>
  <div
    className="d-flex flex-wrap justify-content-center container"
  />
</section>
`;
